File: solutions/csharp/phone-number/PhoneNumber.cs
public class PhoneNumber
{
    public static string Clean(string phoneNumber)
    {
        var f = new string([.. phoneNumber.Where(char.IsDigit)]);

        if (f.Length == 10 &&
            (f[0] - '0' >= 2 ) &&
            (f[0] - '0' <= 9 ) &&
            (f[3] - '0' >= 2 ) &&
            (f[3] - '0' <= 9 )
        )
        {
            return f;
        }
        else if (f.Length == 11 &&
            f[0] == '1' &&
            (f[1] - '0' >= 2 ) &&
            (f[1] - '0' <= 9 ) &&
            (f[4] - '0' >= 2 ) &&
            (f[4] - '0' <= 9 )
        )
        {
            return f.Substring(1, 10);
        }
        else
        {
            throw new ArgumentException("");
        }
    }
}

File: solutions/julia/phone-number/phone-number.jl
function clean(phone_number)

    phone_number = filter(isdigit, phone_number)

    if length(phone_number) == 10 && phone_number[1] ∈ '2':'9' && phone_number[4] ∈ '2':'9'
        return phone_number
    elseif length(phone_number) == 11 && phone_number[1] == '1' && phone_number[2] ∈ '2':'9' && phone_number[5] ∈ '2':'9'
        return phone_number[2:end]
    else
        throw(ArgumentError(""))
    end

end

clean("(223) 456-7890")
